includes/Core/Modules/Modules.php

```diff
@@ -962,17 +968,19 @@ private function get_module_schema() {
 	private function get_active_modules_option() {
 		$option = $this->options->get( self::OPTION_ACTIVE_MODULES );
 
-		if ( is_array( $option ) ) {
-			return $option;
+		if ( ! is_array( $option ) ) {
+			$option = $this->options->get( 'googlesitekit-active-modules' );
 		}
 
-		$legacy_option = $this->options->get( 'googlesitekit-active-modules' );
+		if ( ! is_array( $option ) ) {
+			$option = array();
+		}
 
-		if ( is_array( $legacy_option ) ) {
-			return $legacy_option;
+		if ( in_array( 'analytics', $option, true ) ) {
+			$option[] = 'analytics-4';
```

Comment:
One more thing – we should probably do one or both of these:
- ensure `analytics-4` is not returned by `get_active_modules_option` if  `analytics` is not in the list
- always remove `analytics-4` in `set_active_modules_option` so that `analytics-4` is never set in the option.